service/history/taskPriorityAssigner.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package history
+
+import (
+	"sync"
+
+	workflow "github.com/uber/cadence/.gen/go/shared"
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/cache"
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/log/tag"
+	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/common/quotas"
+)
+
+type (
+	taskPriorityAssigner interface {
+		Assign(queueTask) error
+	}
+
+	taskPriorityAssignerImpl struct {
+		currentClusterName string
+		domainCache        cache.DomainCache
+		config             *Config
+		logger             log.Logger
+		scope              metrics.Scope
+
+		sync.RWMutex
+		rateLimiters map[string]quotas.Limiter
+	}
+)
+
+var _ taskPriorityAssigner = (*taskPriorityAssignerImpl)(nil)
+
+const (
+	taskPriorityActiveTransfer int = iota
+	taskPriorityActiveTimer
+	taskPriorityActiveThrottled
+	taskPriorityReplicationStandby
```

Comment:
From the implementation, it looks like the taskPriorityReplicationStandby used for active replication task or standy timer/ transfer task? shoule it name taskPriorityReplication?